Fix NJT OpInfo entry for nn.functional.prelu

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #144889
* #144588
* #144587
* #144586
* #144585
* #144584
* #144583
* __->__ #144582

Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

The OpInfo entry for prelu was wrong before this PR; `weight` needs to be passed as well. The op isn't fully implemented yet.

jbschlosser committed:
Fix NJT OpInfo entry for nn.functional.prelu

[ghstack-poisoned]

soulitzer commented on torch/testing/_internal/opinfo/definitions/nested.py:
```diff
@@ -1217,6 +1217,31 @@ def sample_inputs_nn_functional_linear(op_info, device, dtype, requires_grad, **
+            device=device,
+            dtype=dtype,
+            requires_grad=requires_grad,
+        )
```
Should we also test the case where weight can be a scalar?
https://pytorch.org/docs/stable/generated/torch.nn.functional.prelu.html

jbschlosser replied:
yes! good call

jbschlosser committed:
Update on "Fix NJT OpInfo entry for nn.functional.prelu"


Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

The OpInfo entry for prelu was wrong before this PR; `weight` needs to be passed as well. The op isn't fully implemented yet.

[ghstack-poisoned]

jbschlosser commented:
@pytorchbot merge
